geode-redis/src/main/java/org/apache/geode/redis/internal/executor/CommandFunction.java

```diff
@@ -16,26 +16,53 @@
 
 package org.apache.geode.redis.internal.executor;
 
+import static org.apache.geode.redis.internal.RedisCommandType.HSET;
+
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.regex.Pattern;
 
 import org.apache.geode.cache.Region;
 import org.apache.geode.cache.execute.Function;
 import org.apache.geode.cache.execute.FunctionContext;
+import org.apache.geode.cache.execute.FunctionService;
+import org.apache.geode.cache.execute.ResultCollector;
 import org.apache.geode.cache.execute.ResultSender;
 import org.apache.geode.internal.cache.execute.RegionFunctionContextImpl;
 import org.apache.geode.redis.internal.ByteArrayWrapper;
 import org.apache.geode.redis.internal.RedisCommandType;
+import org.apache.geode.redis.internal.RedisDataType;
+import org.apache.geode.redis.internal.executor.hash.RedisHash;
 import org.apache.geode.redis.internal.executor.set.RedisSet;
 import org.apache.geode.redis.internal.executor.set.StripedExecutor;
+import org.apache.geode.redis.internal.executor.set.SynchronizedStripedExecutor;
 
 public class CommandFunction implements Function<Object[]> {
```

Comment:
this class is going to grow with each new command added.  Is there any way we can break this up?

Reply:
Probably. I don't have any ideas. But it does seem like we have a bunch of things that are related to each other and maybe they can be refactored in some way. I do like having a single function. For example we could have a function for each command but that would cause a bunch of busy work each time we add function support for another redis command. I would suggest we wait on this refactoring until we get all the commands we want to support for our first release done. Then we can look at those and figure out the best way to organize the code for those commands.